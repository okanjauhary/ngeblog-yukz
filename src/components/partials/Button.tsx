import React from "react";
import { css } from "@emotion/core";
import classNames from "classnames";

type ButtonProp = {
  disabled?: boolean;
  type?: "button" | "submit";
  medium?: boolean;
  large?: boolean;
};

const Button: React.FC<ButtonProp> = props => {
  const { children, disabled = false, type = "button" } = props;
  const buttonClassName = classNames({
    "my-button": true,
    "my-button--disabled": !!disabled,
    "my-button--medium": props.medium,
    "my-button--large": props.large,
  });

  return (
    <div
      className={buttonClassName}
      css={css`
        margin: 10px;
        display: inline-block;
      `}>
      <button
        disabled={disabled}
        type={type}
        css={css`
          outline: none;
          border: none;
          background-color: transparent;
          cursor: pointer;
        `}>
        {children}
      </button>
    </div>
  );
};

export default Button;
