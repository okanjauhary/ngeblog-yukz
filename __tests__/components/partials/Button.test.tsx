import React from "react";
import { render, mount } from "enzyme";
import Button from "../../../src/components/partials/Button";

describe("Button.tsx", () => {
  test("Should render component and match snapshot", () => {
    const wrapper = render(<Button />);

    expect(wrapper).toMatchSnapshot();
  });

  test("Should set button disable correctly", () => {
    const wrapper = mount(<Button disabled />);

    expect(wrapper.exists(".my-button--disabled")).toBe(true);
  });

  test("Should set button type correctly", () => {
    const type = "submit";
    const wrapper = mount(<Button type={type} />);

    const button = wrapper.find(`.my-button button[type="${type}"]`);
    expect(button.exists()).toBe(true);
  });
});
