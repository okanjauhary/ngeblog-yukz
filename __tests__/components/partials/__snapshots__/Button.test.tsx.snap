// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button.tsx Should render component and match snapshot 1`] = `
.emotion-0 {
  margin: 10px;
  display: inline-block;
}

.emotion-1 {
  outline: none;
  border: none;
  background-color: transparent;
  cursor: pointer;
}

<div
  class="my-button emotion-0"
>
  <button
    class="emotion-1"
    type="button"
  />
</div>
`;
